category_encoders/target_encoder.py

```diff
@@ -219,24 +185,25 @@ def _check_dict_key_tuples(dict_to_check: dict[Any, tuple]) -> tuple[bool, int]:
         return min_tuple_size == max_tuple_size, min_tuple_size
 
     def _fit(self, X: util.X_type, y: util.y_type, **kwargs) -> None:
-        if isinstance(self.hierarchy, dict):
+        inverted_hierarchy, self.hierarchy_depth = self._generate_inverted_hierarchy()
```

Comment:
is there a reason for setting `self.hierarchy_depth` instead of just keeping it a variable in the scope of the `_fit` function?

Reply:
`self.hierarchy_depth` is used by `fit_target_encoding` internally.  That function looks like it's part of the public API, but I don't see it explicitly tested in `tests/test_target_encoder.py` and I'm confused how it would work without doing all the setup work handled in `_fit`.

Changing `self.hierarchy_depth` to a local variable scoped in `_fit` would require it to be recomputed in `fit_target_encoding`.